### PR TITLE
Add the test framework

### PR DESCRIPTION
**Pull request by mprahl**

### CODE REVIEW

**Comment by mprahl** [comment]
@mikebonnet  and @yashvardhannanavati could you please review?

File: docker/Dockerfile-tests [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  python2-fedmsg \
+  python3-fedmsg \
+  python3-flake8 \
+  python2-flask \
+  python3-flask \
+  python2-mock \
+  python3-mock \
+  # Until 1.6.0 is supported, we need to hardcode these two entries
+  python2-neo4j-driver-1.5.3 \
+  python3-neo4j-driver-1.5.3 \
```
Comment (yashvardhannanavati) [inline docker/Dockerfile-tests]: Once the [PR](https://github.com/release-engineering/estuary-api/pull/138) in estuary-API is merged, we can remove these right?

**Comment by yashvardhannanavati** [comment]
:+1: 
P.S : I understood most of it but I could have missed a few nuances

File: docker/Dockerfile-tests (hunk 1 above)
Comment (mprahl) [reply docker/Dockerfile-tests]: It turns out neomodel doesn't support python-neo4j-driver 1.6. I filed a PR upstream to fix that.

**Pull request merged**